org-bookmark-heading.el

```diff
@@ -217,7 +217,8 @@ supported, in which case it should be an entry ID)."
             ;; so it won't be selected when the indirect buffer is killed.
             (set-window-prev-buffers nil (append (cdr (window-prev-buffers))
                                                  (car (window-prev-buffers))))))
-        (unless (equal (buffer-file-name (buffer-base-buffer)) filename)
+        (unless (equal (abbreviate-file-name (buffer-file-name (buffer-base-buffer)))
```

Comment:
We should probably compare both the abbreviated and non-abbreviated forms of the filename, otherwise any bookmark created before this change will cause the warning.

Reply:
Indeed, abbreviating the buffer filename was a bad idea. Thanks for pointing out that.

How about expanding the bookmark filename? Except for some strange symlink cases (e.g. IDs are recorded with destination paths, while bookmarks point to origins), it should work.

Reply:
That might do it.  Have you tested that code?  Does it work?

Reply:
Yes, it works on my machine.